packages/rest/src/__tests__/unit/rest.server/rest.server.open-api-spec.unit.ts

```diff
@@ -393,6 +408,137 @@ describe('RestServer.getApiSpec()', () => {
+              content: {
+                'application/json': {
+                  schema: {
+                    title: 'loopback.example',
+                    properties: {
+                      test: {
+                        type: 'string',
+                      },
+                    },
+                  },
+                },
+              },
+            }),
+          )
+          .build(),
+      );
+
+      const EXPECTED_SPEC = anOpenApiSpec()
+        .withOperation(
+          'get',
+          '/',
+          anOperationSpec().withResponse(200, {
+            description: 'Example',
+            content: {
+              'application/json': {
+                schema: {
+                  title: 'loopback.example',
+                  properties: {
+                    test: {
+                      type: 'string',
+                    },
+                  },
+                },
+              },
+            },
+          }),
+        )
+        .build();
+
+      const spec = await server.getApiSpec();
+      expect(spec).to.eql(EXPECTED_SPEC);
+
+      await server.stop();
+    });
+
+    it('runs consolidator if consolidate is set to true', async () => {
+      const options: {rest: RestServerConfig} = {
+        rest: {openApiSpec: {consolidate: true}},
+      };
+      options.rest = givenHttpServerConfig(options.rest);
+      app = new Application(options);
+      app.component(RestComponent);
+      server = await app.getServer(RestServer);
+      await server.start();
+
+      server.api(
+        anOpenApiSpec()
+          .withOperation(
+            'get',
+            '/',
+            anOperationSpec().withResponse(200, {
+              description: 'Example',
+              content: {
+                'application/json': {
+                  schema: {
+                    title: 'loopback.example',
+                    properties: {
+                      test: {
+                        type: 'string',
+                      },
+                    },
+                  },
+                },
+              },
+            }),
+          )
+          .build(),
+      );
+
+      const EXPECTED_SPEC = anOpenApiSpec()
+        .withOperation(
+          'get',
+          '/',
+          anOperationSpec().withResponse(200, {
+            description: 'Example',
+            content: {
+              'application/json': {
+                schema: {
+                  $ref: '#/components/schemas/loopback.example',
+                },
+              },
+            },
+          }),
+        )
+        .withComponents(
+          aComponentsSpec().withSchema('loopback.example', {
+            title: 'loopback.example',
+            properties: {
+              test: {
+                type: 'string',
+              },
+            },
+          }),
+        )
+        .build();
+
+      const spec = await server.getApiSpec();
+      expect(spec).to.eql(EXPECTED_SPEC);
+
+      await server.stop();
+    });
+  });
+
```

Comment:
I think you can add another test for null/undefined to make sure the default value is honored.

Reply:
Hey, Thanks for taking a look. I couldn't figure out how to get the rest config to apply.  I'm not confident if it is either that that the service is bound incorrectly to the rest server or I just cannot form the test correctly.